File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque

def dfs(graph, v, visited):
    visited[v] = True
    print(v, end=' ')
    for i in sorted(graph[v]) :
        if not visited[i] :
            dfs(graph, i, visited)

def bfs(graph, start):
    visited = [False] * (len(graph))
    queue = deque([start])
    visited[start] = True
    while queue:
        v = queue.popleft()
        print(v, end=' ')
        for i in sorted(graph[v]):
            if not visited[i]:
                queue.append(i)
                visited[i] = True

n, m, v = map(int, input().split())
graph = [[] for _ in range(n + 1)]

for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

visited_dfs = [False] * (n + 1)
dfs(graph, v, visited_dfs)
print()

bfs(graph, v)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java

import java.io.*;
import java.util.*;

public class Main {
	static int N, M, V;
	static ArrayList<Integer>[] A;
	static boolean visited[];
	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
		N = sc.nextInt();
		M = sc.nextInt();
		V = sc.nextInt();
		
		visited = new boolean[N + 1];
		A = new ArrayList[N + 1];
		for(int i = 1; i < N + 1; i++) {
			A[i] = new ArrayList<Integer>();
		}
		
		for(int i = 0; i < M; i++) {
			int a, b;
			a = sc.nextInt();
			b = sc.nextInt();
			A[a].add(b);
			A[b].add(a);
		}
		
		for(int i = 1; i <= N; i++) {
			Collections.sort(A[i]); // 오름차순 sorting
		}
		
		DFS(V);
		System.out.println();
		visited = new boolean[N + 1];
		BFS(V);
	}
	private static void BFS(int start) {
		visited[start] = true;
		Queue<Integer> q = new LinkedList<>();
		q.add(start);
		while(!q.isEmpty()) {
			int now = q.poll();
			System.out.print(now + " ");
			for (int next : A[now]) {
				if(!visited[next]) {
					visited[next] = true;
					q.add(next);
				}
			}
		}
		
	}
	private static void DFS(int start) {
		visited[start] = true;
		System.out.print(start + " ");
		for(int i : A[start]) {
			if (!visited[i]) {
				DFS(i);
			}
		}
		
	}
	
	
}

